Neos.Flow/Classes/Http/AbstractMessage.php

```diff
@@ -76,8 +78,9 @@ public function getHeaders()
      * Dates are returned as DateTime objects with the timezone set to GMT.
      *
      * @param string $name Name of the header
-     * @return array|string An array of field values if multiple headers of that name exist, a string value if only one value exists and NULL if there is no such header.
+     * @return string[] An array of field values, may be empty if the header was not set (check with hasHeader)
      * @api
+     * @deprecated Not deprecated but from 5.1 this will only return arrays of strings according to the PSR-7 standard.
```

Comment:
Not true… In the current code (and thus 5.1) this will still return string or array…

Reply:
Correct, that should read "from next major..."